daemon/algod/api/server/v2/utils.go

```diff
@@ -432,7 +440,82 @@ func convertTxnResult(txnResult simulation.TxnResult) PreEncodedSimulateTxnResul
 		AppBudgetConsumed:      omitEmpty(txnResult.AppBudgetConsumed),
 		LogicSigBudgetConsumed: omitEmpty(txnResult.LogicSigBudgetConsumed),
 		TransactionTrace:       convertTxnTrace(txnResult.Trace),
+		UnnamedResources:       convertUnnamedResourceAssignment(txnResult.UnnamedResources),
+	}
+}
+
+func convertUnnamedResourceAssignment(assignment *simulation.ResourceAssignment) *model.SimulationUnnamedResourceAssignment {
+	if assignment == nil {
+		return nil
+	}
+	converted := model.SimulationUnnamedResourceAssignment{
+		MaxTotalRefs: uint64(assignment.MaxTotalRefs),
+		MaxAccounts:  uint64(assignment.MaxAccounts),
+		MaxAssets:    uint64(assignment.MaxAssets),
+		MaxApps:      uint64(assignment.MaxApps),
+		MaxBoxes:     uint64(assignment.MaxBoxes),
+		EmptyBoxRefs: omitEmpty(uint64(assignment.NumEmptyBoxRefs)),
```

Comment:
I think?
```suggestion
		EmptyBoxRefs: omitEmpty(assignment.NumEmptyBoxRefs),
```

Reply:
I think this is tied to `simulation.ResourceAssignment.NumEmptyBoxRefs`, which happen to be an `int`.

But I think it can only be unsigned for the count cannot be negative?

Reply:
Right, the `uint64` cast is needed because `NumEmptyBoxes` is an int, and the only REST integer type we support is `uint64`.

I chose to make `NumEmptyBoxes` optional since it can be 0 in many cases.